src/document.rs

```diff
@@ -314,14 +328,18 @@ impl Document {
                 .unwrap_or(default_excerpt_separator);
 
             let excerpt = if let Some(excerpt_str) = excerpt_attr {
-                excerpt_str
+                excerpt_str.to_string()
             } else if excerpt_separator.is_empty() {
-                ""
+                String::new()
```

Comment:
It's a small thing but you could avoid the allocation here. You could either declare `let mut excerpt` and assign it in the if clauses or you call `try!(self.render_html(..., context, source))` inside each clause. Does that make sense?